src/veriff/veriff.service.ts

```diff
@@ -0,0 +1,171 @@
+import {
+  BadRequestException,
+  Injectable,
+  InternalServerErrorException,
+  Logger,
+  UnauthorizedException,
+} from '@nestjs/common';
+
+import * as crypto from 'crypto';
+import {
+  VerifiableCredentialFactory,
+  VeriffWatchlistService,
+  getTonomyOpsIssuer,
+  getFieldValue,
+  signVerifiableCredential,
+} from './veriff.helpers';
+import { VeriffWebhookPayload, WatchlistScreeningResult } from './veriff.types';
+
+import { util } from '@tonomy/tonomy-id-sdk';
+
+import { CommunicationGateway } from '../communication/communication.gateway';
+export type VeriffPayload = {
+  appName: string;
+};
+
+const ENABLE_PEP_CHECK = false;
+
+@Injectable()
+export class VeriffService {
+  constructor(
+    private readonly credentialFactory: VerifiableCredentialFactory,
+    private readonly veriffWatchlistService: VeriffWatchlistService,
+    private readonly logger: Logger,
+    private readonly communicationGateway: CommunicationGateway,
+  ) {}
+  private readonly VERIFF_SECRET =
+    process.env.VERIFF_SECRET || 'default_secret'; // .env usage
+
+  validateSignature(signature: string, payload: any): boolean {
```

Comment:
always keep type safety when possible. makes developing and debugging much easier.
`validateSignature(signature: string, payload: VeriffWebhookPayload)`